openreview/venue_request/process/matching_pre_process.py

```diff
@@ -1,4 +1,21 @@
 def process(client, note, invitation):
+    _, baseurl_v2 = openreview.tools.get_base_urls(client)
+    client_v2 = openreview.api.OpenReviewClient(baseurl=baseurl_v2, token=client.token)
+    forum_note = client.get_note(note.forum)
+    venue_id = forum_note.content.get('venue_id', '')
+    domain = client_v2.get_group(venue_id)
+    submission_name = domain.content.get('submission_name', '').get('value')
+    senior_area_chairs_name = domain.content.get('senior_area_chairs_name', {}).get('value')
 
-    if 'Yes' in note.content['compute_affinity_scores'] and 'upload_affinity_scores' in note.content:
-        raise openreview.OpenReviewException('Either upload your own affinity scores or select affinity scores computed by OpenReview')
\ No newline at end of file
+    submissions_over_2000 = client_v2.get_notes(invitation=f'{venue_id}/-/{submission_name}', offset=2000)
+    compute_affinity_scores = note.content.get('compute_affinity_scores') == 'Yes'
+
+    matching_group = note.content['matching_group']
+    role_name = matching_group.split('/')[-1]
+    pretty_role = role_name.replace('_', ' ')
+
+    if compute_affinity_scores and 'upload_affinity_scores' in note.content:
```

Comment:
I would leave this if before checking the submission count that is needed later.